libs/sdk-ui-ext/src/internal/dashboardEmbedding/DashboardLayout/utils/sizing.ts

```diff
@@ -467,3 +466,47 @@ export function getDashboardLayoutWidgetDefaultHeight(widgetType: WidgetType, vi
 
     return dimension.defHeightPx;
 }
+
+export function getLayoutWithoutGridHeights<TWidget>(
+    layout: IDashboardLayout<TWidget> | IDashboardLayoutItem<TWidget>[],
+): IDashboardLayout<TWidget> {
```

Comment:
You can simplify it with layout builder-> something like:
```
const layoutBuilder = DashboardLayoutBuilder.for(layout);
return layout.modifySections((section) =>
        section.modifyItems((itemBuilder, itemFacade) => {
            const widget = itemFacade.widget();
            if (isDashboardLayout(widget)) {
                const updatedLayout = getLayoutWithoutGridHeights(widget);
                return itemBuilder.setItem(updatedLayout);
            }

            return itemBuilder.setItem(removeGridHeightFromItemSize);
        }),
    );
```

Reply:
Thanks for the input. Just missing one Type for `updatedLayout`. Have `any` but wasn't able to find the correct one.

Reply:
I see... Only "more exact" typing which comes to my mind now is:
`const updatedLayout = (getLayoutWithoutGridHeights(widget) as unknown) as TWidget;`
So you narrow down `TWidget -> IDashboardLayout<DashboardWidget>`, and then force type it back to `TWidget`